src/variables/utils/buildVarsOption.ts

```diff
@@ -2,8 +2,30 @@
 import {OPTION_NAME} from 'src/variables/constants'
 
 // Types
-import {File, Property} from 'src/types/ast'
-import {VariableAssignment} from 'src/types/ast'
+import {File, Property, VariableAssignment} from 'src/types'
+import {Variable} from 'src/types/variables'
+import {filterUnusedVarsBasedOnQuery} from 'src/shared/utils/filterUnusedVars'
+import {asAssignment} from '../selectors'
+
+export const buildUsedVarsOption = (
+  query: string | string[],
+  allVariables: Variable[],
+  windowVars?: VariableAssignment[]
+): File => {
+  let filteredVars
+  if (Array.isArray(query)) {
+    filteredVars = filterUnusedVarsBasedOnQuery(allVariables, query)
+  } else {
+    filteredVars = filterUnusedVarsBasedOnQuery(allVariables, [query])
+  }
```

Comment:
const 
```suggestion
  const filteredVars = filterUnusedVarsBasedOnQuery(allVariables, Array.isArray(query) ? query : [query])
```

Reply:
or maybe set this prior to the function receiving the query so that all the parameters are input like `[text]` or `text` based on whether they're an array or not?